passes/pmgen/ice40_wrapcarry.cc

```diff
@@ -51,6 +51,7 @@ void create_ice40_wrapcarry(ice40_wrapcarry_pm &pm)
 	cell->setParam("\\LUT", st.lut->getParam("\\LUT_INIT"));
 
 	cell->attributes = std::move(st.carry->attributes);
+	cell->attributes.insert(st.lut->attributes.begin(), st.lut->attributes.end());
```

Comment:
I think this isn't quite right: this won't merge the `\src` attribute (instead it would replace its content with LUT's `\src`), nor the `\keep` attribute (something like `(* keep = 0 *)` is valid as currently recognized by Yosys and equivalent to no `\keep` attribute at all, so `(*keep*) SB_CARRY` + `(*keep=0*) SB_LUT` would result in wrong behavior).

I think we should have something like `AttrObject::merge` that takes care of attributes known to Yosys itself.

Reply:
Another place where `AttrObject::merge_attributes` would be useful is in the `opt_lut` pass.

Reply:
> this won't merge the \src attribute (instead it would replace its content with LUT's \src)

I'm pretty sure the `insert()` semantic is not to overwrite map entries? Indeed, my test shows that the SB_CARRY's `answer = 42` is preserved:
https://github.com/YosysHQ/yosys/blob/a7e0cca48003dbb212326c37c0b8a5580c2e6d2c/tests/arch/ice40/wrapcarry.ys#L25-L50

Regarding `(* keep *)` I will treat this as a special case and OR the two results.

Reply:
> I'm pretty sure the `insert()` semantic is not to overwrite map entries?

Right, sorry, I forgot how maps work in STL and Yosys. But that has the same fundamental problem, which is that the value of `\src` is derived from just one cell instead of both of them.

Reply:
> > I'm pretty sure the `insert()` semantic is not to overwrite map entries?
> 
> Right, sorry, I forgot how maps work in STL and Yosys. But that has the same fundamental problem, which is that the value of `\src` is derived from just one cell instead of both of them.

Yes, we can only pick one or the other and I've given `SB_CARRY` priority over the `SB_LUT4`. I can append the `src` value (separated by `|`)  but that breaks their usual meaning of indicating the progression one particular cell made.

To summarise, the `SB_CARRY` attributes have priority over `SB_LUT4` except in the case of `(* keep = 0 *)`. On unwrapping `SB_CARRY` gets all the attributes and `SB_LUT4` gets none. It sounds like this lossiness is still not acceptable, then I'd have to preserve both sets of attributes separately and create a custom `ice40_wrapcarry -unwrap` pass instead of using a `techmap` call. Does that sound better?

Reply:
> but that breaks their usual meaning of indicating the progression one particular cell made.

I didn't understand that Yosys `\src` attributes have this invariant. In that case I retract my request, and I think the changes here are fine. (I'm not sure if this is a *good* invariant, but this discussion shouldn't block this PR, and I'll raise it elsewhere in the future if I think it's important.)

Reply:
Okay, I'm now preserving all attributes on both `SB_CARRY` and `SB_LUT4` and added a `ice40_wrapcarry -unwrap` option to undo that